managed-pointer/cpp-20/D0567R1.md

```diff
@@ -5,222 +5,384 @@
+future_type_t<Executor, void> put(Executor, ManagedPtrN...);
+template <typename Executor, typename Predicate, typename... ManagedPtrN>
+future_type_t<Executor, void> then_put(Executor, Predicate, ManagedPtrN...);
+template <typename... ManagedPtrN>
+future_type_t<Executor, void> get(ManagedPtrTN...);
+template <typename Predicate, typename... ManagedPtrN>
+future_type_t<Executor, void> then_get(Predicate, ManagedPtrN...);
+template <typename SynchronisationChannel, typename SrcExecutor,
+          typename DestExecutor, typename... ManagedPtrN>
+future_type_t<Executor, void>
+sync(SynchronisationChannel = __default_synchronisation_channel__{},
+     SrcExecutor, DestExecutor, ManagedPtrN...);
+template <typename SynchronisationChannel, typename SrcExecutor,
+          typename DestExecutor, typename Predicate, typename... ManagedPtrN>
+future_type_t<Executor, void>
+then_sync(SynchronisationChannel = __default_synchronisation_channel__{},
+          SrcExecutor, DestExecutor, Predicate, ManagedPtrN...);
+
+/* allocation functions */
+template <typename Executor, typename... ManagedPtrN>
+future_type_t<Executor, void> allocate(Executor, ManagedPtrN...);
+template <typename Executor, typename Predicate, typename... ManagedPtrN>
+future_type_t<Executor, void> then_allocate(Executor, Predicate, ManagedPtrN...);
+template <typename Executor, typename... ManagedPtrN>
+future_type_t<Executor, void> deallocate(Executor, ManagedPtrN...);
+template <typename Executor, typename Predicate, typename... ManagedPtrN>
+future_type_t<Executor, void> then_deallocate(Executor, Predicate,
+                                              ManagedPtrN...);
+
 }  // namespace execution
+}  // namespace concurrency_v2
 }  // namespace experimental
 }  // namespace std
 ```
+*Figure 1: Execution synopsis*
 
-*Figure 1: managed_ptr class template*
+### 3.2. Requirements on Executor and Execution Contex
 
-The ``managed_ptr`` class template has a single template parameter; ``T`` specifying the type of the elements.
+In order to facilitate the `managed_ptr`, it is necessary to introduce requirements on the *executor* and the *execution context*; the class associated with an *executor* which encapsulates the underlying execution resource on which *execution agents* are executed.
 
-The type parameter ``T`` specifies the element type of the container, any ``T`` satisfies the ``managed_ptr`` element type requirements if:
+#### 3.2.1. Execution Context Memory Region
 
-* ``T`` is a standard layout type.
-* ``T`` is copy constructible.
+In addition to a number of *execution agents* an *execution context* must also encapsulate at least one memory region that is concurrently accessible by all *execution agents* of any single invcation of an execution function. It is not required that the memory region be concurrently accessible to *execution agents* of another invocation of an execution function or *execution agents* executing on a different *execution context*.
```

Comment:
Yes this is intentional, if a system has a fine grained unified address space it is possible for two execution agents in different invocations to access the same memory region concurrently. However the `managed_ptr` in it's current state does not permit this and therefore does not require it.